File: problems/2562_find_the_array_concatenation_value/solution.py
from typing import List


class Solution:
    def findTheArrayConcVal(self, nums: List[int]) -> int:
        val = 0
        i, j = 0, len(nums) - 1
        while i < j:
            val += int(str(nums[i]) + str(nums[j]))
            i += 1
            j -= 1
        if i == j:
            val += nums[i]
        return val

File: problems/2562_find_the_array_concatenation_value/Solution.java
class Solution {
    public long findTheArrayConcVal(int[] nums) {
        long val = 0;
        int i = 0, j = nums.length - 1;
        while (i < j) { val += Long.parseLong(nums[i]+""+nums[j]); i++; j--; }
        if (i == j) val += nums[i];
        return val;
    }
}
